sktime/networks/resnet.py

```diff
@@ -0,0 +1,166 @@
+# -*- coding: utf-8 -*-
+"""Residual Network (ResNet) (minus the final output layer)."""
+
+__author__ = ["James Large", "Withington", "nilesh05apr"]
+
+from sktime.networks.base import BaseDeepNetwork
+from sktime.utils.validation._dependencies import (
+    _check_dl_dependencies,
+    _check_soft_dependencies,
+)
+
+_check_soft_dependencies(
+    "keras-self-attention",
+    package_import_alias={"keras-self-attention": "keras_self_attention"},
+    severity="warning",
+)
+
+
+class ResNetNetwork(BaseDeepNetwork):
+    """
+    Establish the network structure for a ResNet.
+
+    Adapted from the implementations used in [1]
+
+    Parameters
+    ----------
+    random_state : int, optional (default = 0)
+        The random seed to use random activities.
+
+    Notes
+    -----
+    Adpated from the implementation source code
+    https://github.com/hfawaz/dl-4-tsc/blob/master/classifiers/resnet.py
+
+    References
+    ----------
+    .. [1] H. Fawaz, G. B. Lanckriet, F. Petitjean, and L. Idoumghar,
+
+    Network originally defined in:
+
+    @inproceedings{wang2017time, title={Time series classification from
+    scratch with deep neural networks: A strong baseline}, author={Wang,
+    Zhiguang and Yan, Weizhong and Oates, Tim}, booktitle={2017
+    International joint conference on neural networks (IJCNN)}, pages={
+    1578--1585}, year={2017}, organization={IEEE} }
+
+    """
+
+    def __init__(self, random_state=0):
+        _check_dl_dependencies(severity="error")
+        _check_soft_dependencies(
+            "keras-self-attention",
+            package_import_alias={"keras-self-attention": "keras_self_attention"},
+            severity="error",
+        )
+        super(ResNetNetwork).__init__()
+        self.random_state = random_state
+
+    def build_model(self, input_shape, **kwargs):
```

Comment:
The abstract method that all estimators implement is `build_network()`. This minor reason is why tests are failing across all runners. A small typo :)